### PR TITLE
Use .afterFirstUnlock accessibility

### PR DESCRIPTION
**Pull request by eoger**
Fixes https://github.com/mozilla/application-services/issues/3083.

### CODE REVIEW

File: components/fxa-client/ios/FxAClient/FxAccountStorage.swift [hunk 1]
```diff
@@ -8,13 +8,17 @@ import SwiftKeychainWrapper
...
 
     init(keychainAccessGroup: String?) {
         keychainWrapper = KeychainWrapper.sharedAppContainerKeychain(keychainAccessGroup: keychainAccessGroup)
     }
 
     func read() -> FxAccount? {
-        if let json = keychainWrapper.string(forKey: KeyChainAccountStorage.keychainKey) {
+        // Firefox iOS v25.0 shipped with the default accessibility, which breaks Send Tab when the screen is locked.
+        // This method migrates the existing keychains to the correct accessibility.
+        keychainWrapper.ensureStringItemAccessibility(KeyChainAccountStorage.accessibility, forKey: KeyChainAccountStorage.keychainKey)
```
Comment (eoger) [inline components/fxa-client/ios/FxAClient/FxAccountStorage.swift:20]: This does the migration of current states.

File: components/fxa-client/ios/FxAClient/FxAccountStorage.swift [hunk 2]
```diff
@@ -26,14 +30,37 @@ class KeyChainAccountStorage {
...
 
     func clear() {
-        if !keychainWrapper.removeObject(forKey: KeyChainAccountStorage.keychainKey) {
+        if !keychainWrapper.removeObject(forKey: KeyChainAccountStorage.keychainKey, withAccessibility: KeyChainAccountStorage.accessibility) {
             FxALog.error("Could not clear account state.")
         }
     }
 }
+
+public extension KeychainWrapper {
```
Comment (eoger) [inline components/fxa-client/ios/FxAClient/FxAccountStorage.swift:45]: This was copy-pasted from firefox-ios code.

File: components/fxa-client/ios/FxAClient/FxAccountStorage.swift [hunk 3]
```diff
@@ -8,13 +8,23 @@ import SwiftKeychainWrapper
 class KeyChainAccountStorage {
     internal var keychainWrapper: KeychainWrapper
     internal static var keychainKey: String = "accountJSON"
+    internal static var accessibility: KeychainItemAccessibility = .afterFirstUnlock
```
Comment (rfk) [inline components/fxa-client/ios/FxAClient/FxAccountStorage.swift:11]: IIUC this changes *all* FxA state to be accessible after the first unlock, including the sessionToken and sync encryption keys. Does this risk weakening our security story compared to pre-v25.0 releases of Firefox iOS?

**Comment by rfk** [comment]
The [docs for kSecAttrAccessibleAfterFirstUnlock](https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlock) say:

```
Items with this attribute migrate to a new device when using encrypted backups.
```

Which kind of surprised me, but I guess it's fine - if you're restoring from a backup you could reasonably expect that your signed-in state would be preserved, and I don't think anything will break.

File: components/fxa-client/ios/FxAClient/FxAccountStorage.swift (hunk 3 above)
Comment (eoger) [reply components/fxa-client/ios/FxAClient/FxAccountStorage.swift:11]: It shouldn't, Firefox iOS used to store (and still does) Sync and FxA state under this accessibility.

**Pull request merged**